src/bitcoin-cli.cpp

```diff
@@ -713,6 +715,9 @@ static UniValue ConnectAndCallRPC(BaseRequestHandler* rh, const std::string& str
     UniValue response(UniValue::VOBJ);
     // Execute and handle connection failures with -rpcwait.
     const bool fWait = gArgs.GetBoolArg("-rpcwait", false);
+    const int timeout = gArgs.GetArg("-rpcwaittimeout", DEFAULT_WAIT_CLIENT_TIMEOUT);
+    const int64_t deadline = GetTime<std::chrono::microseconds>().count() + timeout;
```

Comment:
I think you want this to be `std::chrono::seconds` instead of `std::chrono::microseconds`, or to change the doc above to mention the parameter is passed in microseconds.